### PR TITLE
fix: dice yaml replace * error

### PR DESCRIPTION
**Pull request by CeerDecy**
#### What this PR does / why we need it:

fix dice yaml replace * error

#### Which issue(s) this PR fixes:

- Fixes #your-issue_number
- [Erda Cloud Issue Link](https://erda.cloud/erda/dop/projects/387/issues/gantt?id=632619&iterationID=-1&pId=0&type=BUG)

#### Specified Reviewers:

/assign @iutx 

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |      fix dice yaml replace * error        |
| 🇨🇳 中文    |      修复dice yaml 替换 * 后导致反序列化错误        |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by iutx — DISMISSED** [review]
LGTM @sfwn

File: pkg/parser/diceyml/diceyml.go [hunk 1]
```diff
@@ -280,7 +281,12 @@ func (d *DiceYaml) getEnvValueData(env ...string) ([]byte, error) {
 		} else {
 			value = findValue
 		}
-		return []byte(strings.TrimSpace(string(value)))
+		value = []byte(strings.TrimSpace(string(value)))
+		if specialValueRegex.Match(value) {
+			value = append(value, '\'')
```
Comment (sfwn) [inline pkg/parser/diceyml/diceyml.go:286]: use `fmt.Sprintf("'%s'")` directly for better understanding.

**Review by sfwn — CHANGES_REQUESTED** [review]

**Review by sfwn — APPROVED** [review]

**Pull request merged**